enterprise/views.py

```diff
@@ -452,22 +482,44 @@ def get(self, request):
 
         # Get enterprise_customer to start in case we need to render a custom 404 page
         # Then go through other business logic to determine (and potentially overwrite) the enterprise customer
-        enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
+        try:
+            enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
+        except Http404:
+            error_code = 'ENTGDS008'
+            log_message = (
+                '[Enterprise DSC Flow] Received incorrect value for enterprise customer in request. '
```

Comment:
Should we just say `Could not find enterprise customer with provided data` or something like that?